src/storage/data_table.cpp

```diff
@@ -68,11 +69,18 @@ DataTable::SlotIterator &DataTable::SlotIterator::operator++() {
   common::SpinLatch::ScopedSpinLatch guard(&table_->blocks_latch_);
```

Comment:
We should defer acquiring this latch until absolutely necessary (i.e. referencing the vector on line 77).  Everything else is unnecessary execution in a critical section and adds potential for extra branch mispredictions while holding the latch.